src/main/java/com/selfxdsd/selfweb/api/ContractsApi.java

```diff
@@ -581,27 +581,25 @@
                         final Wallet wallet = project.wallets().active();
                         payment = wallet.pay(found);
                     }
-                    final Invoice active = contract.invoices().active();
                     if(payment == null) {
+                        final Invoice active = contract.invoices().active();
                         resp = ResponseEntity.ok(
                             Json.createObjectBuilder()
                                 .add("paid", found.invoiceId())
                                 .add("active", new JsonInvoice(active))
                                 .build()
                                 .toString()
                         );
-                    } else {
+                    } else if (payment.status()
+                        .equals(Payment.Status.SUCCESSFUL)) {
+                        final Invoice active = contract.invoices().active();
                         resp = ResponseEntity.ok(
                             Json.createObjectBuilder()
                                 .add("paid", found.invoiceId())
                                 .add(
                                     "payment",
                                     Json.createObjectBuilder()
-                                        .add("status", payment.status())
                                         .add(
-                                            "failReason",
-                                            payment.failReason()
-                                        ).add(
```

Comment:
@amihaiemil Since we don't send the fail message to client anymore, I've removed `status` and `failReason` from payment json. Payment response for an active Invoice can be only succesfull otherwise the response status is 400.

Reply:
@criske No, we have to send the fail message to client. There are 2 types of failure status: FAILED (no cash on card, no payment method active, no payout method etc) and ERROR (when all the preconditions are met but Stripe threw an exception).

So please don't change this logic on the backend endpoint. Just a change in the front-end (change the message to a generic one, if the status is ERROR) is enough :D